include/glow/Graph/Graph.h

```diff
@@ -460,23 +460,44 @@ class Function final : public Named {
-  /// offsets provide individual quantization parameters for each filter group
-  /// in \p filter. \p kernels defines the size of the height and width
+  /// convolves the 4D \p input with \p filter and \p bias. \p filterScales and
+  /// \p filterOffsets provide individual quantization parameters for each
+  /// filter group in \p filter while \p biasScales and \p biasOffsets provide
+  /// individual quantization parameters for each bias element corresponding to
+  /// each output channel. \p kernels defines the size of the height and width
   /// dimensions of the filters. \p strides defines the number of steps to take
   /// in the input for each output cell. \p pads defines how many zero padding
   /// cells should be added to the input during convolution. \p group defines
   /// the number of groups the input and output channels should be divided into
-  /// and convolved separately. If bias is FloatTy then it will be quantized
-  /// to Int32QTy automatically.
-  /// NOTE: ChannelwiseQuantizedConvolutionNode does
-  /// not yet have an implementation so attempting to run a graph containing
-  /// this node fails.
+  /// and convolved separately. \p dilation defines the filter dilation.
+  /// This function is flexible and has the following features:
+  /// - it can be provided with a floating-point \p filter and the function will
+  ///   quantize automatically the filter channelwise using the given schema
+  ///   \p schema and type \p filterElemQTy.
+  /// - it can be provided with a floating-point \p bias and the function will
+  ///   quantize automatically the bias channelwise using the given schema
+  ///   \p schema and type \p biasElemQTy.
+  /// - if \p filter is floating-point and \p filterScales or \p filterOffsets
+  ///   are not provided then this function will derive them automatically.
+  /// - if \p filter is quantized then \p filterScales or \p filterOffsets are
+  ///   mandatory.
+  /// - if \p bias is floating-point and \p biasScales or \p biasOffsets are not
+  ///   provided then this function will derive them automatically.
+  /// - if \p bias is quantized  and \p biasScales or \p biasOffsets are not
+  ///   provided then this function will assume the implicit parameters
+  ///   biasScales[i] = inputScale * filterScales[i] and biasOffsets[i] = 0.
+  ///   To be noted that this case can handle safely only INT32 bias data type
+  ///   because for INT8 type the bias will almost certainly be saturated.
+  /// This function requires \p filter and \p bias operands to be constants.
   ChannelwiseQuantizedConvolutionNode *createChannelwiseQuantizedConv(
       llvm::StringRef name, NodeValue input, NodeValue filter, NodeValue bias,
-      NodeValue scales, NodeValue offsets, TypeRef outTy,
-      llvm::ArrayRef<unsigned_t> kernels, llvm::ArrayRef<unsigned_t> strides,
-      llvm::ArrayRef<unsigned_t> pads, unsigned_t group);
+      NodeValue filterScales, NodeValue filterOffsets, NodeValue biasScales,
```

Comment:
The channelwise (element-wise) quantized bias has the property that it is exactly representable in the quantized domain (no quantization loss) as long the scale parameters are chosen properly, even when using bias types lower than Int32QTy:
- By "chosen properly" I mean that although the implicit selection `biasScales[i] = inputScale * filterScales[i]` is natural for Int32QTy, for lower precision types it will almost certainly saturate the bias data (when quantized by Glow from float to that respective type like Int8Qty for example) and the unit tests actually prove that. And bias quantization is very important since the bias values directly impact the output samples of the convolution, even more so than the weight values.
- Therefore for lower bias precision types the implementation must be slightly more flexibile/generic and NOT assume the implicit scales `biasScales[i] = inputScale * filterScales[i]`.
- I also don't like having assumptions embedded in the inner core of an implementation (like we had for the Interpreter) since that constraints the usage. That is why I moved the logic of choosing the "proper" bias quantization parameters in `Graph.cpp`. That way all the backends can benefit from that logic (and not just the Interpreter) like the CPU for which I also added an implementation
- In terms of consistency I also think the ChannelwiseQuantizedConvolution should have an implementation similar to the others (same bias treatment as normal quantized convolution, etc).
- Finally, the logic I added in `Graph.cpp` is flexible enough so that when dealing with Int32QTy bias, if not provided with explicit bias scales parameters it will select the implicit scale parameters and the numerical behavior will be the same as now.
Sorry for the long explanation, but the point I'm trying to make is that the Glow contributors (like myself) might deal with different use cases and we might want future-proof some of the implementations by making them more generic.
Thanks!